nesta_ds_utils/networks/build.py

```diff
@@ -53,11 +58,51 @@ def build_coocc(
 
     # edge weights are all times a pair of tokens have co-occured in the same sequence
     edge_weights = _cooccurrence_counts(sequences)
+
+    # if using similarity metrics as edge attributes, calculate those
+    if "jaccard_similarity" in edge_attributes:
+        jaccard_similarity = _jaccard_similarity(edge_weights, all_tokens)
+    if "association_strength" in edge_attributes:
+        association_strength = _association_strength(edge_weights, all_tokens)
+    if "cosine" in edge_attributes:
+        cosine_sim = _cosine_sim(edge_weights, all_tokens)
+    if "inclusion_index" in edge_attributes:
+        inclusion_index = _inclusion_index(edge_weights, all_tokens)
+
+    # add edges to network
     for key, val in edge_weights.items():
         weight = {"weight": val} if weighted else {}
-        network.add_edge(key[0], key[1], **weight)
+        jaccard_sim = (
+            {"jaccard_similarity": jaccard_similarity[key]}
+            if "jaccard_similarity" in edge_attributes
+            else {}
+        )
+        assoc_str = (
+            {"association_strength": association_strength[key]}
+            if "association_strength" in edge_attributes
+            else {}
+        )
+        co_sim = {"cosine": cosine_sim[key]} if "cosine" in edge_attributes else {}
+        inclusion = (
+            {"inclusion_index": inclusion_index[key]}
+            if "inclusion_index" in edge_attributes
+            else {}
+        )
+
+        network.add_edge(
+            key[0], key[1], **weight, **jaccard_sim, **assoc_str, **co_sim, **inclusion
+        )
```

Comment:
Having a look at the nx docs, I'm not sure `weight` actually has any special significance, but is just the default attribute name for a weight. In any case, I think this could be done without all the conditionals and just by building and unpacking a single dict and using `add_edges_from`. Something like:

1. Create a set of 'records' for each node where the inclusion of an edge attribute is triggered by conditionals on the arguments of the function. Should end up with something structured `[(0, 1 {"frequency": 4, "jaccard": 0.5}), (0, 3, {"frequency": 10, "jaccard": 0.2}), ...]`
2. Pass this into `g.add_edges_from`.

Alternatively, you could iterate through `edge_weights.items()` as you do here and construct the dictionary of attributes on the fly and then simply do

```python
network.add_edge(key[0], key[1], **weights)
if directed:
    network.add_edge(key[1], key[0], **weights)
```

Reply:
n.b. a benefit to doing either of these is that you could get rid of one of the function args and just include edge frequency weight as one of the weight options